src/Interpreters/castColumn.cpp

```diff
@@ -51,4 +51,14 @@ ColumnPtr castColumnAccurateOrNull(const ColumnWithTypeAndName & arg, const Data
     return castColumn<CastType::accurateOrNull>(arg, type);
 }
 
+ColumnPtr tryCastColumnAccurate(const ColumnWithTypeAndName & arg, const DataTypePtr & type)
+try
```

Comment:
This introduces exceptions in a normal code path, which we don't allow.

Reply:
Better to do `castColumnAccurateOrNull` and check for NULL.

Reply:
But this is exactly what we did before...

Reply:
We should make `castColumnAccurateOrNull` to work properly for `LowCardinality`.

Reply:
> We should make `castColumnAccurateOrNull` to work properly for `LowCardinality`.

But what type will it be? And what should we do for Arrays which also cannot be inside Nullable?

Reply:
I guess we should use `castFieldAccurateOrNull` here.

Reply:
Ok.

Reply:
Also check `convertFieldToType`.

Reply:
This becomes overly complicated.

1. We cannot use `convertFieldToType` because the Field type system is inaccurate. 
2. We cannot use castFieldAccurateOrNull because some types cannot be inside nullable.
3. We cannot introduce exceptions in a normal code path, because we don't allow.

Now I wonder, why can't we allow every type be inside Nullable?

Related issues:
https://github.com/ClickHouse/ClickHouse/issues/38738
https://github.com/ClickHouse/ClickHouse/issues/37881
https://github.com/ClickHouse/ClickHouse/issues/36894
https://github.com/ClickHouse/ClickHouse/issues/38786
https://github.com/ClickHouse/ClickHouse/issues/31440

and some more issues related with `join_uses_nulls`

Reply:
I'll make `castColumnAccurateOrNull` to work properly for `LowCardinality` so that it fixes the related bug report. It's unlikely that user would use other non-nullable types as primary key. So maybe it's enough.